File: goapi/wallet.go
package main

import (
	"time"
	"fmt" 
)

type Wallet struct {
	balance int
}

func NewWallet() *Wallet {
	return &Wallet{
		balance: 0,
	}
}
 
func (b *Wallet) Balance() int {
	fmt.Println("b.balance ", b.balance )
	return b.balance
}

func (b *Wallet) Deposit(amount int) {
	time.Sleep(time.Millisecond * 100)
	fmt.Println("Incrementou ", b.balance, amount )
	b.balance += amount
}

File: java/src/main/java/com/example/demo/Wallet.java
package com.example.demo;

import java.util.concurrent.TimeUnit;

public class Wallet {

    int balance;

    public Wallet() {
        balance = 0;
    }

    public int Balance() {
       return balance;
    }

    public void Deposit(int amount) throws InterruptedException {
        TimeUnit.MILLISECONDS.sleep(100);
        balance += amount;
        System.out.println("Incrementou ");
    }

}
